### DISCUSSION TITLE
AI needs a delay/toggle on closing doors and hatches

### DISCUSSION THREAD

**Discussion by Rattle189** [FakeFishGames/Barotrauma#7334] [Ideas and suggestions]
Can the AI have a delay before they close doors/hatches when they pass it? I keep getting stuck and sometimes the crew also gets stuck when they climb the ladders and an AI crewmate closes the hatch. It's also so damn annoying trying to climb ladders or run to a room to repair something when the AI just closes the hatch/door on you

**Comment by MaaiG** [comment]
> delay/toggle on closing doors and hatches

...and [problem with opening](https://github.com/Regalis11/Barotrauma/discussions/7424)! 
I think it might be nice if AI will have a delay between trying to open the door! Or number of attempts will be limited and odd )))

In my case, the bot can’t use my airlock camera.
After pressing the button you have to wait for the airlock chamber to fill water - then the door will open.
The bot presses the button, but the door does not open immediately. The bot presses the button again and the airlock is shuts down. 
Then the bot constantly tries to press the button (as if in a panic) and gets into the loop. The sequence does not work because the bot constantly switches the button and interrupts it.

**Comment by itchyOwl** [comment]
Addressed in Regalis11/Barotrauma-development@e8a33a1 (next unstable).
